### PR TITLE
document conv

### PR DESCRIPTION
**Pull request by felixrehren**
Cf. https://github.com/JuliaLang/julia/issues/20539

> Convolves two vectors. (If `u`, `v` are the coefficients of two polynomials, `conv(u, v)`
are the coefficients of the product of these polynomials.) Uses FFT algorithm.

Is it appropriate to use `conv(u, v)` as a plural "are" the coefficients? Both "is" and "are" feels wrong

### CODE REVIEW

File: base/dsp.jl [hunk 1]
```diff
@@ -121,9 +121,10 @@ function deconv{T}(b::StridedVector{T}, a::StridedVector{T})
 end
 
 """
-    conv(u,v)
+    conv(u, v)
 
-Convolution of two vectors. Uses FFT algorithm.
+Convolves two vectors. (If `u`, `v` are the coefficients of two polynomials, `conv(u, v)`
```
Comment (Sacha0) [inline base/dsp.jl]: Perhaps "(If `u` and `v` contain the coefficients of two polynomials, `conv(u, v)` yields the coefficients of the product of those polynomials.)"?

**Review by Sacha0 — APPROVED** [review]

**Comment by kshyatt** [comment]
It seems we have some extra commits here? Is that the case?

**Comment by felixrehren** [comment]
@kshyatt Yes, sorry for the cruft. Still trying to figure out how to gracefully update my fork. Is there something I can do to eliminate these extra commits, or can they be squashed in merge?

File: base/dsp.jl [hunk 2]
```diff
@@ -103,9 +103,9 @@ function _filt_fir!(out, b, x, si, col)
 end
 
 """
-    deconv(b,a) -> c
+    deconv(b, a) -> c
 
-Construct vector `c` such that `b = conv(a,c) + r`.
+Construct vector `c` such that `b = conv(a, c) + r`.
 Equivalent to polynomial division.
```
Comment (KristofferC) [inline base/dsp.jl]: Not necessary for this PR but places like this are excellent for doctests.

**Comment by Sacha0** [comment]
> Is there something I can do to eliminate these extra commits, or can they be squashed in merge?

Perhaps check out this branch (`git checkout docconv`), back it up (`git branch docconvbackup1`), then interactively rebase dropping the first two commits and squash/fixup the last commit into the third (`git rebase -i HEAD~4`, replace `pick` with `drop` for the first two commits and `pick` with `fixup` or `squash` for the last commit in the text description of the rebase that pops up, then save and close the description)? (If something goes wrong, you can revert the `docconv` branch to master, and then cherry-pick the relevant commits from your backup branch onto `docconv`.) Best!

File: base/dsp.jl [hunk 3]
```diff
@@ -121,9 +121,11 @@ function deconv{T}(b::StridedVector{T}, a::StridedVector{T})
 end
 
 """
-    conv(u,v)
+    conv(u, v)
 
-Convolution of two vectors. Uses FFT algorithm.
+Convolves two vectors. (If `u` and `v` describe the coefficients of two polynomials,
```
Comment (ararslan) [inline base/dsp.jl]: Typically in describing functionality in docstrings, we stick to the imperative. So this should be "convolve" instead of "convolves."

**Review by Sacha0 — APPROVED** [review]

**Review by ararslan — APPROVED** [review]

**Comment by tkelman** [comment]
reopen against DSP.jl?

**Comment by ararslan** [comment]
I'm going to close this since `conv` is no longer in Base. A PR to DSP.jl would be fantastic!

**Pull request closed without merging**